src/plugins/kibana_usage_collection/server/collectors/config_usage/register_config_usage_collector.ts

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { UsageCollectionSetup } from '../../../../usage_collection/server';
+import { ConfigUsageData, CoreUsageDataStart } from '../../../../../core/server';
+
+export function registerConfigUsageCollector(
+  usageCollection: UsageCollectionSetup,
+  getCoreUsageDataService: () => CoreUsageDataStart
+) {
+  const collector = usageCollection.makeUsageCollector<ConfigUsageData | undefined>({
+    type: 'kibana_config_usage',
+    isReady: () => typeof getCoreUsageDataService() !== 'undefined',
+    /**
+     * No schema for this collector.
+     * This collector will collect non-default configs from all plugins.
+     * Mapping each config to the schema is inconvenient for developers
+     * and would result in 100's of extra field mappings.
+     *
+     * We'll experiment with flattened type and runtime fields before comitting to a schema.
+     */
+    schema: {},
```

Comment:
Can we create an issue to follow up on this with `DYNAMIC_KEY` or something else we might want to implement?

Reply:
https://github.com/elastic/kibana/issues/97599

Note that `DYNAMIC_KEY` will not work here since the value is also dynamic (keyword or number or boolean)